docs/content/connectors/mongodb-cdc.md

```diff
@@ -0,0 +1,440 @@
+
+- For Debezium's MongoDB Connector, it read the `oplog.rs` collection of each replicaset's master node.
+- For MongoDB's Kafka Connector, it subscribe `Change Stream` of MongoDB.
+
+MongoDB's `oplog.rs` collection doesn't keep the changed record's update before state, so it's hard to extract the full document state by a single `oplog.rs` record and convert it to change log stream accepted by Flink (Insert Only, Upsert, All).
+Additionally, MongoDB 5 (released in July) has changed the oplog format, so the current Debezium connector cannot be used with it.
+
+**Change Stream** is a new feature provided by MongoDB 3.6 for replica sets and sharded clusters allows applications to access real-time data changes without the complexity and risk of tailing the oplog.
+Applications can use change streams to subscribe to all data changes on a single collection, a database, or an entire deployment, and immediately react to them.
+
+**Lookup Full Document for Update Operations** is a feature provided by **Change Stream** which can configure the change stream to return the most current majority-committed version of the updated document. Because of this, we can easily collect the latest full document and convert the change log to Flink's **Upsert Changelog Stream**. 
+
+By the way, Debezium's MongoDB change streams exploration mentioned by [DBZ-435](https://issues.redhat.com/browse/DBZ-435) is on roadmap. 
+If it's done, we can support both for users to choose.
+
+### DataStream Source
+
+The MongoDB CDC connector can also be a DataStream source. You can create a SourceFunction as the following shows:
+
+```java
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.source.SourceFunction;
+import com.ververica.cdc.debezium.StringDebeziumDeserializationSchema;
+import com.ververica.cdc.connectors.mongodb.MongoDBSource;
+
+public class MongoDBSourceExample {
+    public static void main(String[] args) throws Exception {
+        SourceFunction<String> sourceFunction = MongoDBSource.<String>builder()
+                .hosts("localhost:27017")
+                .username("flink")
+                .password("flinkpw")
+                .database("inventory")
+                .collection("products")
+                .deserializer(new StringDebeziumDeserializationSchema())
+                .build();
+
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+        env.addSource(sourceFunction)
+                .print().setParallelism(1); // use parallelism 1 for sink to keep message ordering
+
+        env.execute();
+    }
+}
+```
+
+
+Data Type Mapping
+----------------
+
+<div class="wy-table-responsive">
+<table class="colwidths-auto docutils">
+    <thead>
+      <tr>
+        <th class="text-left">BSON type<a href="https://docs.mongodb.com/manual/reference/bson-types/"></a></th>
```

Comment:
Explain a bit why we map from BSON type at the front?

Reply:
Fixed.